modulus/distributed/manager.py

```diff
@@ -359,24 +359,24 @@ def setup(
             else:
                 manager._local_rank = local_rank
 
+        manager._device = torch.device(
+            f"cuda:{manager.local_rank}" if torch.cuda.is_available() else "cpu"
+        )
+
+        if manager._distributed:
             # Setup distributed process group
             dist.init_process_group(
```

Comment:
there is an option to set `device_id` for `init_process_group`

> a single, specific device to “bind” this process to, allowing for backend-specific optimizations. Currently this has two effects, only under NCCL: the communicator is immediately formed (calling ncclCommInit* immediately rather than the normal lazy call) and sub-groups will use ncclCommSplit when possible to avoid unnecessary overhead of group creation. If you want to know NCCL initialization error early, you can also use this field.

Reply:
Probably not, the way the documentation is written, I actually think that this should be the default anyways.

Reply:
checked it again, it is only available from PyTorch 2.3.0 onwards, if possible, the code now set this id